### PR TITLE
Add error propogation information to output

### LINKED ISSUE #66: Add propagated error to the registration.txt output
Current registration.txt output ends with

```
RMSEs:
X = +/-0.488,
Y = +/-0.805,
Z = +/-0.861,
3D = +/-1.276
```

If we assume 3m spherical error of the photogrammetrically derived foundation DSMs, then the total error is computed as

```
Total Error = SQRT(3^2 + 3D_RMSE^2)
```

In the case above, that equates to 3.26 m.

We had discussed potentially breaking that into horizontal and vertical components, but that can be added at a later time. (We'd either need a provider to specify their estimated horizontal and vertical error, or we could break down the spherical down based on some Gaussian assumptions.)

Maybe this is best shown as a new block, summarizing the assumptions and equations used, e.g., 

```
PROPAGATED ERROR
----------------
Assumed global 3D error = +/-3
3D_RMSE = +/-1.276
Total Error is computed as SQRT( global_3D_error^2 + 3D_RMSE^2 )
Total Error = +/-3.260
```

### PR DESCRIPTION
**Pull request by j9ac9k**
Fixes #66

### CODE REVIEW

File: src/codem/registration/dsm.py [hunk 1]
```diff
@@ -445,6 +446,13 @@ def _output(self) -> None:
                 f"\nZ = +/-{self.rmse_xyz[2]:.3f},"
                 f"\n3D = +/-{self.rmse_3d:.3f}"
             )
+            f.write("\nPropagated Error:")
+            f.write(
+                "\nAssuemed global 3D error = +/-3m"
```
Comment (chambbj) [inline src/codem/registration/dsm.py:451]:
```suggestion
                "\nAssumed global 3D error = +/-3m"
```

File: src/codem/registration/icp.py [hunk 2]
```diff
@@ -485,4 +486,11 @@ def _output(self) -> None:
                 f"\nZ = +/-{self.rmse_xyz[2]:.3f},"
                 f"\n3D = +/-{self.rmse_3d:.3f}"
             )
+            f.write("\nPropagated Error:")
+            f.write(
+                "\nAssuemed global 3D error = +/-3m"
```
Comment (chambbj) [inline src/codem/registration/icp.py:491]:
```suggestion
                "\nAssumed global 3D error = +/-3m"
```

File: src/codem/registration/icp.py [hunk 3]
```diff
@@ -485,4 +486,11 @@ def _output(self) -> None:
...
+                f"\n3D_RMSE = +/-{math.pow(self.rmse_3d, 2):.3f}"
```
Comment (chambbj) [inline src/codem/registration/icp.py:492]: No need to square.

File: src/codem/registration/icp.py [hunk 4]
```diff
@@ -485,4 +486,11 @@ def _output(self) -> None:
...
+                "\nTotal Error is computed as √(global_3d_error²+3D_RMSE²)"
+                f"\nTotal Error = {math.hypot(3, self.rmse_3d)}"
```
Comment (chambbj) [inline src/codem/registration/icp.py:494]:
```suggestion
                f"\nTotal Error = +/-{math.hypot(3, self.rmse_3d):.3f}"
```

**Review by chambbj — CHANGES_REQUESTED** [review]
A few suggestions and comments provided.

File: src/codem/registration/dsm.py [hunk 5]
```diff
@@ -445,6 +446,13 @@ def _output(self) -> None:
...
+                f"\n3D_RMSE = +/-{math.pow(self.rmse_3d, 2):.3f}"
```
[the remark this answers is not in the document]
Comment (j9ac9k) [reply src/codem/registration/dsm.py:452]: Because I misread your example in the issue :)

**Pull request merged**